whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapRequestValidator.java

```diff
@@ -0,0 +1,69 @@
+package net.ripe.db.whois.api.rdap;
+
+import net.ripe.db.whois.api.fulltextsearch.FullTextIndex;
+import net.ripe.db.whois.common.dao.RpslObjectDao;
+import net.ripe.db.whois.common.ip.IpInterval;
+import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.ObjectType;
+import net.ripe.db.whois.common.rpsl.attrs.AttributeParseException;
+import net.ripe.db.whois.common.rpsl.attrs.AutNum;
+import net.ripe.db.whois.common.rpsl.attrs.Domain;
+import net.ripe.db.whois.common.source.SourceContext;
+import net.ripe.db.whois.query.acl.AccessControlListManager;
+import net.ripe.db.whois.query.planner.AbuseCFinder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import static net.ripe.db.whois.common.rpsl.ObjectType.ORGANISATION;
+
+@Component
+public class RdapRequestValidator {
```

Comment:
You could make this a "Mapper" class instead, and return the parsed value if you need it afterwards.

Also include as block here.

Reply:
This class is for only validating the key supplied by user.  If supplied key maps to object type it is valid. we are not using parsed key as this code ultimately calls the same code that query search api calls.
regarding including AS_Block, user cannot request for  as_block as its not allowed.
User can only pass autnum not as block. so validation can happen only on autnum.

Reply:
Ok!